query/graphql/src/main/java/org/codice/ddf/graphql/servlet/GraphQLTransformerServlet.java

```diff
@@ -0,0 +1,126 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.graphql.servlet;
+
+import com.google.common.cache.Cache;
+import com.google.common.cache.CacheBuilder;
+import com.google.common.cache.RemovalCause;
+import com.google.common.cache.RemovalNotification;
+import graphql.servlet.GraphQLProvider;
+import graphql.servlet.OsgiGraphQLHttpServlet;
+import java.util.List;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import org.codice.ddf.admin.api.Events;
+import org.codice.ddf.admin.api.FieldProvider;
+import org.codice.ddf.graphql.transform.GraphQLTransformCommons;
+import org.osgi.service.event.Event;
+import org.osgi.service.event.EventHandler;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class GraphQLTransformerServlet extends OsgiGraphQLHttpServlet implements EventHandler {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(GraphQLTransformerServlet.class);
+  private static final long CACHE_EXPIRATION_IN_SECONDS = 1;
+  private static final long CACHE_CLEANUP_INVOCATION_IN_SECONDS = 1;
+
+  private static final String BINDING_FIELD_PROVIDER = "GraphQL servlet binding field provider %s";
+  private static final String UNBINDING_FIELD_PROVIDER =
+      "GraphQL servlet unbinding field provider %s";
+
+  private final Cache<String, Object> cache;
+  private final ScheduledExecutorService scheduler;
+  private GraphQLProvider graphQLProvider;
+  private List<FieldProvider> fieldProviders;
+
+  public GraphQLTransformerServlet() {
+    super();
+    setExecutionStrategyProvider(new ExecutionStrategyProviderImpl());
+    setErrorHandler(new GraphQLErrorHandlerImpl());
+    setInstrumentationProvider(new QueryValidationInstrumentationProvider());
+
+    cache =
+        CacheBuilder.newBuilder()
+            .expireAfterWrite(CACHE_EXPIRATION_IN_SECONDS, TimeUnit.SECONDS)
+            .removalListener(this::refreshSchemaOnExpire)
+            .build();
+
+    scheduler = Executors.newScheduledThreadPool(1);
+    scheduler.scheduleAtFixedRate(
+        cache::cleanUp,
+        CACHE_CLEANUP_INVOCATION_IN_SECONDS,
+        CACHE_CLEANUP_INVOCATION_IN_SECONDS,
+        TimeUnit.SECONDS);
+  }
+
+  @Override
+  public void destroy() {
```

Comment:
❓ Should we shutdown according to the javadoc here? https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/ExecutorService.html

Reply:
✏️ We can use `MoreExecutors.shutdownAndAwaitTermination` from google guava. https://github.com/google/guava/blob/master/guava/src/com/google/common/util/concurrent/MoreExecutors.java

Reply:
I read through the javadoc some more. Were you referring shutting down like they do here? 
```
void shutdownAndAwaitTermination(ExecutorService pool) {
   pool.shutdown(); // Disable new tasks from being submitted
   try {
     // Wait a while for existing tasks to terminate
     if (!pool.awaitTermination(60, TimeUnit.SECONDS)) {
       pool.shutdownNow(); // Cancel currently executing tasks
       // Wait a while for tasks to respond to being cancelled
       if (!pool.awaitTermination(60, TimeUnit.SECONDS))
           System.err.println("Pool did not terminate");
     }
   } catch (InterruptedException ie) {
     // (Re-)Cancel if current thread also interrupted
     pool.shutdownNow();
     // Preserve interrupt status
     Thread.currentThread().interrupt();
   }
 }
```
The scheduler is responsible for responding to groups of requests to refresh the graphql schema. If the servlet is being destroyed, I don't think there is any reason we would want to wait for this to finish.